app/config/config.yml

```diff
@@ -66,3 +66,15 @@ swiftmailer:
     username:  "%mailer_user%"
     password:  "%mailer_password%"
     spool:     { type: memory }
+
+# Can be changed directly in the bundle is we found an agreement
+dunglas_action:
+    autodiscover:
+        enabled:   true
+        directories:
+            action: [ ]
+            console: [ Command ]
+    directories:         # List of directories relative to the kernel root directory containing classes to auto-register.
+        action: [ '../src/AppBundle/Controller' ] # Still an autowiring_type definition to add in Symfony to avoid this
```

Comment:
I agree with most of your arguments and I've made changes accordingly in the bundle:

* The "autodiscover" config option have been removed, only user bundles are now scanned by default, it's safer in better on all front (dunglas/dunglasActionBundle#26)
* Standard `Controller` and `Command` are now the default config (dunglas/dunglasActionBundle#26) - it allows to be as close as possible to current Symfony conventions
* The `action-annotation` routing type is being removed, the usual `annotation` will work (dunglas/DunglasActionBundle#29)

I'm not confortable with adding some magic in the router to guess the service corresponding to the type. The router supports services for a while using the 'service_name:method' syntax for while. It's what use the bundle. I prefer to keep it as is it will be easier to debug.

For the missing autowiring_type, it's for `KernelInterface` but it's not related anymore to this PR.